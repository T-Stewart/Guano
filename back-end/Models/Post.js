const mongoose = require('mongoose')

const PostSchema = new mongoose.Schema({
    thought: Object,
    opinion: String,
    comments: Object,
    userId: String,
    userName: String
})

const Post = mongoose.model('post', PostSchema)

module.exports = Post;