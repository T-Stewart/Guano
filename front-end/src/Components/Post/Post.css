.post-page {
  background: #25272e;
  margin-top: 25px;
  margin-left: 25px;
  margin-right: 25px;
  height: 100vh;
  border-radius: 25em;
  display: grid;
  justify-content: center;
  align-content: center;
  min-height: 900px;
}

.container {
  display: grid;
  justify-content: center;
  align-content: center;
}

.submit-group {
  border: 2px solid black;
  margin-top: 2em;
  width: 75%;
}
.opinion-header {
  font-family: monospace;
}

.opinion-area {
  display: grid;
  justify-content: center;
  align-content: center;
  width: 400px;
  height: 120px;
  color: #cde9ff;
  font-weight: normal;
  background-color: transparent;
  border: solid #cde9ff 1px;
  background-color: rgba(63, 63, 63, 0.255);
}

.edit-opinion-area {
  width: 200px;
  height: 50;
  color: #cde9ff;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  font-weight: normal;
  background-color: transparent;
  border: solid #cde9ff 2px;
}

.form-area {
  display: grid;
  justify-content: center;
  align-content: center;
  margin: 1em;
}

.thought-area {
  color: #cde9ff;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  text-decoration: underline;
  font-weight: normal;
  background-color: transparent;
}

::placeholder {
  color: #cde9ff;
  font-style: italic;
}

.post-area {
  color: #cde9ff;
  font-family: "Lucida Sans", "Lucida Sans Regular", "Lucida Grande",
    "Lucida Sans Unicode", Geneva, Verdana, sans-serif;
  font-weight: normal;
  border: solid #cde9ff 1px;
  background-color: #33333341;
  text-align: center;
  border-radius: 1em;
  margin: 1em;
  width: 200px;
  height: 100px;
}

.opinion-header {
  text-align: center;
  color: #cde9ff;
  text-decoration: underline;
  font-family: monospace;
}

.post-return {
  background-color: transparent;
  font-weight: bolder;
}
#thought {
  text-decoration: underline;
}

.gen-btn {
  width: 400px;
  height: 100px;
  font-size: 200%;
  color: #24262d;
  margin: 10px;
  border-radius: 32px;
  background-color: #9ca6c9;
  font-family: monospace;
  text-align: center;
  cursor: pointer;
  transition: all 0.1s ease-in-out;
}
button:focus {
  outline: 0;
}

.return-btns {
  display: flex;
  font-weight: lighter;
  background-color: transparent;
  justify-content: center;
}

.delete-btn,
.edit-btn,
.cancel-btn,
.submit-btn {
  width: 70px;
  height: 25px;
  background: #cde9ff;
  border-radius: 5px;
  color: #24262d;
  font-family: "Lucida Sans", "Lucida Sans Regular", "Lucida Grande",
    "Lucida Sans Unicode", Geneva, Verdana, sans-serif;
  text-align: center;
  cursor: pointer;
  transition: all 0.1s ease-in-out;
  box-shadow: 6px 6px -10px black(255, 255, 255, 0.8);
}
.edit-btn,
.delete-btn {
  margin-bottom: 25px;
  margin-left: 3px;
  margin-right: 3px;
}

@media screen and (max-width: 768px) {
  .gen-btn {
    width: 150px;
    height: 70px;
    font-size: 150%;
    border-radius: 200px;
  }
  .opinion-area {
    width: 300px;
    height: 100px;
    background-color: transparent;
    border: solid #25272e 2px;
    background-color: rgba(63, 63, 63, 0.255);
  }
  .opinion-header {
    font-family: monospace;
  }
}
