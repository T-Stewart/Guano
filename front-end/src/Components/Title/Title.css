.title-container {
  background-color: transparent;
  margin-left: 50px;
  margin-right: 50px;
}

.header-text {
  text-align: right;
  color: #bfc3ce;
  margin-right: 20px;
  font-family: "Lucida Sans", "Lucida Sans Regular", "Lucida Grande",
    "Lucida Sans Unicode", Geneva, Verdana, sans-serif;
}

.guano {
  text-align: right;
  margin-right: 20px;
  padding-left: 100px;
  color: #767c91;
  font-family: monospace;
}

@media screen and (max-width: 768px) {
  .header-text,
  .opinion-header {
    text-align: center;
    color: #bfc3ce;
    font-family: "Lucida Sans", "Lucida Sans Regular", "Lucida Grande",
      "Lucida Sans Unicode", Geneva, Verdana, sans-serif;
    margin-right: 0px;
    margin-left: 0px;
  }
  .guano {
    text-align: center;
    margin-right: 20px;
    color: #767c91;
    font-family: monospace;
    font-size: 500%;
    margin-right: 0px;
    padding: 0px;
  }
}
