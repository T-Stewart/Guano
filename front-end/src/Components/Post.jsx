import React from "react"
import $, { event } from "jquery"
import axios from "axios"
import "./Post.css"
import thoughts from "./ThoughtGenerator"


export default class Posts extends React.Component {
    constructor(){
        super();
        this.state = {
            thought: [],
            opinion: '',
            posts: [],
            updateOpinion: '',
            updateThought: '',
            isInEditMode: false,
            isInCommentMode: false,
            updateID:''
         }

    }

    randomThought = () => {
      
        const randomThought = thoughts[Math.floor(Math.random()* thoughts.length)]
       
        this.setState({thought: randomThought})
        
  }

    componentDidMount = () => {
        console.log("Are we mounted?")
        this.getBlogPost();
    }

    getBlogPost = () => {
        axios.get('/api')
        .then((response) => {
            const data = response.data
            this.setState({posts: data})
            
        })
        .catch(() => {
         console.log('Error')
        })
    };

    handleChange = (e) => {
        const target = e.target;
        const name = target.name;
        const value = target.value;

        this.setState({
            [name]: value
        });
    };

    submit = (e) => {
        e.preventDefault();
        // const { match: { params } } = this.props;

        const post = {
            thought: this.state.thought,
            opinion: this.state.opinion
        };

        axios ({
            url: '/api',
            method: 'POST',
            data: post
        })
        .then (() => {
            console.log('Data has been sent to the server')
        })
        .finally(() => {
            this.resetUserInputs();
            this.getBlogPost();
            this.displayPosts(this.state.posts);
        })
        .catch(() => {
            console.log('Error')
        })
    }

    update = (e) => {
        e.preventDefault();
        this.setState({isInEditMode: true})
        const post_id = {
            id: e.target.dataset.id
        } 
        axios ({
            url: '/api/retrieve',
            method:'POST',
            data: post_id
        }) 
        .then((response) => {
            this.setState({updateOpinion: response.data.opinion, updateThought: response.data.thought, updateID: response.data._id})
        })
        .catch(() => {
            console.log('Error')
        })
      }

    save = (e) => {
          e.preventDefault();
          this.setState({isInEditMode: false})
          const post = {
              id: this.state.updateID,
              opinion: this.state.updateOpinion,
              thought: this.state.updateThought
          }
          axios({
              url: '/api/update',
              method: 'POST',
              data: post
          })
          .then((response) =>{
              console.log(response.data)
          })
          .finally(() => {
              this.resetUserInputs();
              this.getBlogPost();
              this.displayPosts(this.state.posts)
          })
          .catch(() => {
              console.log('Error')
          })
      }

    delete = (e) => {
          e.preventDefault()
          const post_id = {
              id: e.target.dataset.id
          }
          axios({
              url: '/api/delete',
              method: 'POST',
              data: post_id
          })
          .then(() => {
              console.log('Data has been deleted')
          })
          .finally(() => {
              this.getBlogPost();
              this.displayPosts(this.state.posts)
          })
          .catch(() => {
              console.log('Error')
          })
      }

    cancelUpdate = (e) => {
        e.preventDefault();
        this.setState({isInEditMode: false})
    }

    // showComment = (e) => {
    //     e.preventDefault()
    //     this.setState({isInCommentMode: true})
    // }
  

 

    resetUserInputs = () => {
    this.setState({
       thought: [],
       opinion: '',
       updateMessage: '',
       updateID: ''
     });
   };


   displayPosts =(posts) => {
      
       if (!posts.length) return null;
       return posts.map((post, index) => (
           <div key={index} className="post-area">
                 <h4 className="post-return" id="thought">{post.thought}</h4>
                 <p className="post-return">{post.opinion}</p>  
       
            <div className="return-btns">
              <form data-id={post._id} onSubmit={ this.delete }>
                <input className="delete-btn" type="submit" value="Delete"/>
              </form>
              <form data-id={post._id} onSubmit={ this.update}>
                <input className="edit-btn" type="submit" value="Edit"/>
              </form>
            </div>  
               </div>
       ))
   };

   render(){
       console.log('State: ', this.state)
       if(this.state.isInEditMode === false) {
      $("#edit-posts").hide()
    }else{
      $("#edit-posts").show()
    }  if(this.state.isInEditMode === true) {
      $("#edit-thought").hide()
    }   if(this.state.isInCommentMode === false) {
        $("#comment-box").hide()
    }else{
        $("#comment-box").show()
    }
   

     return(
         <div className="post-page">  
             <div className="container">
                   <button className="gen-btn" onClick={this.randomThought}>Thought Generator</button>
                    <h3 
                        className="thought-area"
                        name="thought"
                        >{this.state.thought}</h3>
                <form className="form-area" onSubmit={this.submit}>
                  
                    <textarea
                        className="opinion-area"
                        name="opinion"
                        placeholder="What's your opinion"
                        cols="30"
                        rows="3"
                        maxLength="150"
                        value={this.state.opinion}
                        onChange={this.handleChange}>
                  </textarea>                     
                   <input className="submit-btn" type= "submit" value="Submit"/>
                </form>
                
              </div>
                    <h2 className="opinion-header">Your Opinions</h2>
            <div className="container">
                 <div className="edit-container" id="edit-posts">
                     <h4 className="opinion-header"> Edit your post below... </h4>
                        <form className="form-area" onSubmit={this.save}>        
                        <input id="edit-thought" className="edit-opinion-area" type="text" value={this.state.updateThought} onChange={this.handleChange} name="updateThought"/>
                        <textarea
                        name="updateOpinion"
                        placeholder="Change Your mind"
                        className="edit-opinion-area"
                        maxLength="150"
                        value={this.state.updateOpinion}
                        onChange={this.handleChange}>
                        </textarea>
                        <div className="return-btns">
                        <input  className="submit-btn" type="submit" value="Submit"/> 
                        <input className="cancel-btn" type="submit" value="Cancel" onSubmit={this.cancelUpdate}></input>   
                        </div>          
                    </form>
             </div>    
              {this.displayPosts(this.state.posts)}
               
            </div>
                  
        </div>
     )
   };
};

