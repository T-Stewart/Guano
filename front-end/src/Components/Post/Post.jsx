import React from "react"
import $  from "jquery"
import axios from "axios"
import "./Post.css"
import thoughts from "../ThoughtGenerator"
import Title from "../Title/Title"


export default class Posts extends React.Component {
    constructor(){
        super();
        this.state = {
            thought: [],
            opinion: '',
            posts: [],
            updateOpinion: '',
            updateThought: '',
            isInEditMode: false,
            isInCommentMode: false,
            updateID:'',
            firstName: 'tom'
         };
    };

    randomThought = () => {
        const randomThought = thoughts[Math.floor(Math.random()* thoughts.length)]
        this.setState({thought: randomThought}) 
    };

    componentDidMount = () => {
        this.takeUserId();
        console.log("Are we mounted?", this.state.firstName)
        this.getBlogPost();
    };

    getBlogPost = () => {
        axios.get('/api/posts')
        .then((response) => {
            const data = response.data
            this.setState({posts: data}) 
        })
        .catch(() => {
        //  console.log('Error')
        });
    };

    handleChange = (e) => {
        const target = e.target;
        const name = target.name;
        const value = target.value;
        this.setState({
            [name]: value
        });
    };

    submit = (e) => {
        e.preventDefault();
        const post = {
            thought: this.state.thought,
            opinion: this.state.opinion,
            firstName: this.state.firstName,  
        };

        axios ({
            url: '/api/posts',
            method: 'POST',
            data: post
        })
        .then (() => {
            console.log('Data has been sent to the server')
        })
        .finally(() => {
            this.resetUserInputs();
            this.getBlogPost();
            this.displayPosts(this.state.posts);
        })
        .catch(() => {
            console.log('Error')
        });
    };

    update = (e) => {
        e.preventDefault();
        this.setState({isInEditMode: true})
        const post_id = {
            id: e.target.dataset.id
        } 
        axios ({
            url: '/api/posts/retrieve',
            method:'POST',
            data: post_id
        }) 
        .then((response) => {
            this.setState({updateOpinion: response.data.opinion, updateThought: response.data.thought, updateID: response.data._id})
        })
        .catch(() => {
            console.log('Error')
        });
      };

    save = (e) => {
          e.preventDefault();
          this.setState({isInEditMode: false})
          const post = {
              id: this.state.updateID,
              opinion: this.state.updateOpinion,
              thought: this.state.updateThought,
              firstName: this.state.firstName
            }
          axios({
              url: '/api/posts/update',
              method: 'POST',
              data: post
          })
          .then((response) =>{
              console.log(response.data)
          })
          .finally(() => {
              this.resetUserInputs();
              this.getBlogPost();
              this.displayPosts(this.state.posts)
          })
          .catch(() => {
              console.log('Error')
          });
      };

    delete = (e) => {
          e.preventDefault()
          const post_id = {
              id: e.target.dataset.id
          }
          axios({
              url: '/api/posts/delete',
              method: 'POST',
              data: post_id
          })
          .then(() => {
              console.log('Data has been deleted')
          })
          .finally(() => {
              this.getBlogPost();
              this.displayPosts(this.state.posts)
          })
          .catch(() => {
              console.log('Error')
          });
      };

    cancelUpdate = (e) => {
        e.preventDefault();
        this.setState({isInEditMode: false})
    };
  
    resetUserInputs = () => {
    this.setState({
       thought: [],
       opinion: '',
       updateMessage: '',
       updateID: ''
     });
   };

   displayPosts =(posts) => {      
       if (!posts.length) return null;
       return posts.map((post, index) => (
           <div key={index} className="post-area">
                 <h4 className="post-return" id="thought">{post.thought}</h4>
                 <p className="post-return">{post.opinion}</p>  
                <div className="return-btns">
                <form data-id={post._id} onSubmit={ this.delete }>
                    <input className="delete-btn" type="submit" value="Delete"/>
                </form>
                <form data-id={post._id} onSubmit={ this.update}>
                    <input className="edit-btn" type="submit" value="Edit"/>
                </form>
                </div>  
            </div>
       ));
   };

   takeUserId = () => {
    const { params } = this.props
    console.log("hey we are in takeUserid function and we are getting ");
    axios({
      url: '/api/user/find',
      method: 'POST',
      data: params
    }).then((response) => {
      let firstName = response.data.firstName
      this.setState({firstName: firstName})
      console.log(this.state.firstName)
    });
  };

   render(){
       console.log('State: ', this.state)
       if(this.state.isInEditMode === false) {
      $("#edit-posts").hide()
    }else{
      $("#edit-posts").show()
    }  if(this.state.isInEditMode === true) {
      $("#edit-thought").hide()
    }   if(this.state.isInCommentMode === false) {
        $("#comment-box").hide()
    }else{
        $("#comment-box").show()
    } if(this.state.firstName !== undefined) {
      console.log(this.state.firstName )
      console.log("true")
      $(".topsection").show()
    }else{
      console.log('false')
      $(".topsection").hide()
    }
   
     return(
         <div >
            <Title/>
            <div className="post-page" data-testid="post-page-test">  
                <div className="container">
                    <button className="gen-btn" onClick={this.randomThought}>Thought Generator</button>
                    <h3 
                        className="thought-area"
                        name="thought"
                        >{this.state.thought}</h3>
                    <form className="form-area" onSubmit={this.submit}>
                    <textarea
                        className="opinion-area"
                        name="opinion"
                        placeholder="What's your opinion"
                        cols="30"
                        rows="3"
                        maxLength="150"
                        value={this.state.opinion}
                        onChange={this.handleChange}>
                    </textarea>                     
                    <input className="submit-btn" type= "submit" value="Submit"/>
                    </form>
                </div>
                    <h2 className="opinion-header">Your Opinions</h2>
                    <div className="edit-container" id="edit-posts">
                        <h4 className="opinion-header"> Edit your post below... </h4>
                        <form className="form-area" onSubmit={this.save}>        
                            <input id="edit-thought" className="edit-opinion-area" type="text" value={this.state.updateThought} onChange={this.handleChange} name="updateThought"/>
                            <textarea
                            name="updateOpinion"
                            placeholder="Change Your mind"
                            className="edit-opinion-area"
                            maxLength="150"
                            value={this.state.updateOpinion}
                            onChange={this.handleChange}>
                            </textarea>
                            <div className="return-btns">
                                <input  className="submit-btn" type="submit" value="Submit"/> 
                                <input className="cancel-btn" type="submit" value="Cancel" onSubmit={this.cancelUpdate}></input>   
                            </div>          
                        </form>
                </div>    
                    <div className="container">
                    {this.displayPosts(this.state.posts)}     
                    </div> 
            </div>
        </div>
     )
   };
};

