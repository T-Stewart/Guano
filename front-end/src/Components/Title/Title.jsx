import React from "react"
import './Title.css'

export default class Title extends React.Component{
    render(){
        return(
      <div className="title-container" data-testid="title-test">
        <h1 className="guano">Guano</h1>
        <h2 className="header-text">Your Opinion Does Not Matter</h2>
        <h4 className="header-text"> But tell us anyway </h4>
      </div>
       );
    };
};