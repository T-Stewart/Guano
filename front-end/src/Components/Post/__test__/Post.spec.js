import React from'react';
import Post from '../Post'
import { shallow, mount } from 'enzyme'
import Title from '../../Title/Title'

describe('Post', () => {
    let wrapper;

    beforeEach(() => wrapper = shallow(<Post />));

    it('Should render Correctly', () => expect(wrapper).toMatchSnapshot());

    it('Should render a <div />', () => {
        expect(wrapper.find('div').length).toEqual(6)
    })

    it('should render Title components', () => {
        expect(wrapper.containsAllMatchingElements([
            <Title />
        ])).toEqual(true)
    });


});

describe('mounted posts', () => {
    let wrapper;

    beforeEach(() => wrapper = mount(<Post />));

    it('generates random thought', () => {
        const spy = jest.spyOn(wrapper.instance(), 'randomThought');
        wrapper.instance().forceUpdate();
        expect(spy).toHaveBeenCalledTimes(0)
        wrapper.find('.gen-btn').first().simulate('click');
        expect(spy).toHaveBeenCalledTimes(1);
    })
    it('submits thought and opinion on submit', () => {
        const spy = jest.spyOn(wrapper.instance(), 'submit');
        wrapper.instance().forceUpdate();
        expect(spy).toHaveBeenCalledTimes(0)
        wrapper.find('.form-area').first().simulate('submit');
        expect(spy).toHaveBeenCalledTimes(1);
    })
})
