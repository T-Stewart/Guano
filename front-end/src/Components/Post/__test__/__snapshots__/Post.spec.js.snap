// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Should render Correctly 1`] = `ShallowWrapper {}`;
